### PR TITLE
Bug 1828964: Adjust kubevirt modals footer buttons

### PR DESCRIPTION
**Pull request by yaacov**
The CNV details view modals do not align with what OCP has. We should adjust them to align them visually. CNV has set out to align better with Patternfly 4 but we should in the short term ensure we have a similar experience to the rest of the console.

This BZ align kubevirt plugin modal:

  - [x]  Do not include an "X" to close
  - [x]  Align actions to the right
  - [x]  Use the modifier .pf-m-secondary for the cancel action
  - [x]  Use a spacer between the footer actions (32px)

Note: In the case of boot-order modal we also need to top align the modal, this is not easily done for PF4 modal, IMHO we can wait until reset of the console will migrate to PF4, or find a fix that is PF4 ready, before fixing that.

Screenshots:
Before
screenshot-console-openshift-console apps ostest test metalkube org-2020 04 30-14_27_01
After:
screenshot-localhost_9000-2020 04 30-14_27_40

screenshot-localhost_9000-2020 04 30-15_54_44

screenshot-localhost_9000-2020 04 30-15_54_09

### CODE REVIEW

**Comment by yaacov** [comment]
@suomiy @glekner @pcbailey @irosenzw please review

**Comment by yaacov** [comment]
@matthewcarleton please review

**Comment by matthewcarleton** [comment]
@yaacov so for the cd-rom, boot order modals it looks like they are aligned to the middle of the viewport still. Is this a difficult thing to fix?

**Comment by yaacov** [comment]
@matthewcarleton hi,

> for the cd-rom, boot order modals it looks like they are aligned to the middle of the viewport still

only boot-order is in the middle of the screen [1].

> Is this a difficult thing to fix?

Yes, boot-order modal is based on PF4 , I did not find prop or css that can easily move the modal to top
https://www.patternfly.org/v4/documentation/react/components/modal

[1] screenshot of whole screen with modal:
screenshot-localhost_9000-2020 04 30-16_01_27

File: frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx [hunk 1]
```diff
@@ -74,28 +72,29 @@ export const ModalFooter: React.FC<ModalFooterProps> = ({
...
-      {submitButtonText}
-    </Button>
-    <Button variant={ButtonVariant.link} onClick={onCancel} id={prefixedID(id, 'cancel')}>
-      {cancelButtonText}
-    </Button>
+
+    <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+      <Button
+        type="button"
+        variant="secondary"
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx:87]: please use the typed version of variant

File: frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx [hunk 2]
```diff
@@ -74,28 +72,29 @@ export const ModalFooter: React.FC<ModalFooterProps> = ({
...
+        data-test-id="modal-cancel-action"
+        onClick={onCancel}
+      >
+        {cancelButtonText}
+      </Button>
+      <Button variant="primary" isDisabled={isDisabled} id="confirm-action" onClick={onSubmit}>
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx:93]: same here

**Comment by matthewcarleton** [comment]
> @matthewcarleton hi,
> 
> > for the cd-rom, boot order modals it looks like they are aligned to the middle of the viewport still
> 
> only boot-order is in the middle of the screen [1].
> 
> > Is this a difficult thing to fix?
> 
> Yes, boot-order modal is based on PF4 , I did not find prop or css that can easily move the modal to top
> https://www.patternfly.org/v4/documentation/react/components/modal
> 
> [1] screenshot of whole screen with modal:
> screenshot-localhost_9000-2020 04 30-16_01_27

ah ya it's the bullseye layout that's being used so it would require some custom CSS to adjust it.

File: frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx]: :+1: fiexd

File: frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx (hunk 2 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/modals/modal/modal-footer.tsx]: :+1: fiexd

**Pull request merged**